src/fragments/Slider.react.js

```diff
@@ -18,6 +18,30 @@ export default class Slider extends Component {
             : ReactSlider;
         this._computeStyle = computeSliderStyle();
         this.state = {value: props.value};
+        this.syncInputWithSlider = this.syncInputWithSlider.bind(this);
+        this.input = React.createRef();
+    }
+
+    syncInputWithSlider() {
+        if (this.input.current.value > this.props.max) {
+            this.input.current.value = this.props.max;
+        }
+
+        if (this.input.current.value < this.props.min) {
+            this.input.current.value = this.props.min;
+        }
+
+        if (this.timeout) {
+            clearTimeout(this.timeout);
+        }
+
+        const valueAsNumber = Number(this.input.current.value);
```

Comment:
@jdamiba looking good! Playing with this using `pytest -k slsl008 --pause` - a little bit of behavior we might want to tweak: It's possible to delete the text in the input box (in which case it gets treated as zero)
<img width="150" alt="Screen Shot 2021-04-21 at 4 27 06 PM" src="https://user-images.githubusercontent.com/2678795/115616658-737d2780-a2be-11eb-98e3-01f21a1a065c.png">

or enter a value that's incompatible with the `step`
<img width="212" alt="Screen Shot 2021-04-21 at 4 26 55 PM" src="https://user-images.githubusercontent.com/2678795/115616679-79730880-a2be-11eb-98b9-e776aa580031.png">


Also: if you enter an out-of-range number, after the timeout time it will be replaced by the closest limit value. This is a problem if the limit is bigger than zero, it can prevent you from typing in the number you want. For example set `min=10, max=20` and try to type 13 - if you're quick you can get it in various ways, but if you're slower than the timeout you'll end up with 10, then you might type the 3 making 103 which gets pushed to 20.

In all of these cases I think we should allow the value to stay in the input box until blur, and NOT update the slider or the prop. Then on blur my gut reaction is:
- if you cleared the value, reset to the previous prop value. There's also a case where you type a partial number that's not a number, for example "2e" is allowed because "2e3" etc is valid. Also "." - seems like all of these should be treated the same way, reset to the previous prop value.
- if you're out of range, push to the closest limit
- if you're between steps, round to the nearest step

Finally, the reason I commented on this particular line: you're casting to number here, but you already used it above as though it was a number, when it was still a string.